### PR TITLE
[node-manager] update nodeUser resource

### PR DESCRIPTION
**Pull request by RomanenkoDenys**
## Description

This PR adds:
1. Support of node-group selector for node-users.
2. Support of multiple ssh keys per user.
3. Immediate change of corresponding system user, if nodeuser crd is changed.

## Changelog entries

```changes
module: node-manager
type: feature
description: "Update nodeuser resource to support node-group selector and multiple ssh keys."
note:
```

### CODE REVIEW

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go [hunk 1]
```diff
@@ -422,3 +454,188 @@ func (c *BashibleContext) saveChecksum(name, newChecksum string) {
...
+
+	go informer.Run(ctx.Done())
+
+	// Wait for the first sync of the informer cache, should not take long
+	if !cache.WaitForCacheSync(ctx.Done(), informer.HasSynced) {
+		klog.Fatalf("unable to sync caches: %v", ctx.Err())
+	}
+}
+
+func (c *BashibleContext) runBufferedEmitter() chan struct{} {
```
Comment (yalosev) [inline modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go:529]: Maybe we can move this emitter to some common code to not copy the code between different packages

**Comment by EvgenySamoylov** [comment]
Closes #107 and #85

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go (hunk 1 above)
Comment (RomanenkoDenys) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go]: its a good point, i refactor this code.

Comment (RomanenkoDenys) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/context.go]: good point, i refactor them

File: modules/040-node-manager/images/bashible-apiserver/pkg/template/node_user.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+	if nu.PasswordHash != newSpec.PasswordHash {
+		return false
+	}
+
+	if nu.IsSudoer != newSpec.IsSudoer {
+		return false
+	}
+
+	if !slicesIsEqual(nu.NodeGroups[:], newSpec.NodeGroups[:]) {
```
Comment (yalosev) [inline modules/040-node-manager/images/bashible-apiserver/pkg/template/node_user.go:62]: why do you use [:] here? Don't think that it's necessary.

Comment (RomanenkoDenys) [reply modules/040-node-manager/images/bashible-apiserver/pkg/template/node_user.go]: fixed

**Review by yalosev — APPROVED** [review]

**Review by nabokihms — APPROVED** [review]

**Pull request merged**